tests/test_connector.rs

```diff
@@ -0,0 +1,351 @@
+    String::from_utf8(balance.unwrap())
+        .unwrap()
+        .parse()
+        .unwrap()
+}
+
+fn total_supply_eth(master_account: &UserAccount) -> u128 {
+    let balance = master_account.view(CONTRACT_ACC.to_string(), "ft_total_supply_eth", &[]);
+    String::from_utf8(balance.unwrap())
+        .unwrap()
+        .parse()
+        .unwrap()
+}
+
+#[test]
+fn test_near_deposit_balance_total_supply() {
+    let (master_account, _contract) = init(CUSTODIAN_ADDRESS);
+    call_deposit_near(&master_account);
+
+    let balance = get_near_balance(&master_account, DEPOSITED_RECIPIENT);
+    assert_eq!(balance, DEPOSITED_AMOUNT - DEPOSITED_FEE);
+
+    let balance = get_near_balance(&master_account, CONTRACT_ACC);
+    assert_eq!(balance, DEPOSITED_FEE);
+
+    let balance = total_supply(&master_account);
+    assert_eq!(balance, DEPOSITED_AMOUNT);
+
+    let balance = total_supply_near(&master_account);
+    assert_eq!(balance, DEPOSITED_AMOUNT);
+
+    let balance = total_supply_eth(&master_account);
+    assert_eq!(balance, 0);
+}
+
+#[test]
+fn test_deposit_eth_and_near() {
+    let (master_account, _contract) = init(CUSTODIAN_ADDRESS);
+    call_deposit_near(&master_account);
+    let (master_account, _contract) = init(EVM_CUSTODIAN_ADDRESS);
+    call_deposit_eth(&master_account);
+}
+
+#[test]
+fn test_eth_deposit_balance_total_supply() {
+    let (master_account, _contract) = init(EVM_CUSTODIAN_ADDRESS);
+    call_deposit_eth(&master_account);
+
+    let balance = get_eth_balance(&master_account, validate_eth_address(RECIPIENT_ETH_ADDRESS));
+    assert_eq!(balance, DEPOSITED_EVM_AMOUNT - DEPOSITED_EVM_FEE);
+
+    let balance = total_supply(&master_account);
+    assert_eq!(balance, DEPOSITED_EVM_AMOUNT);
+
+    let balance = total_supply_eth(&master_account);
+    assert_eq!(balance, DEPOSITED_EVM_AMOUNT);
+
+    let balance = total_supply_near(&master_account);
+    assert_eq!(balance, 0);
+}
+
+#[test]
+fn test_withdraw_near() {
+    let (master_account, _contract) = init(CUSTODIAN_ADDRESS);
+    call_deposit_near(&master_account);
+
+    let withdraw_amount = 100;
+    let res = master_account.call(
+        CONTRACT_ACC.to_string(),
+        "withdraw_near",
+        json!({
+            "recipient_id": RECIPIENT_ETH_ADDRESS,
+            "amount": withdraw_amount,
+        })
+        .to_string()
+        .as_bytes(),
+        DEFAULT_GAS,
+        0,
+    );
+    res.assert_success();
+
+    let balance = get_near_balance(&master_account, DEPOSITED_RECIPIENT);
+    assert_eq!(
+        balance,
+        DEPOSITED_AMOUNT - DEPOSITED_FEE - withdraw_amount as u128
+    );
+
+    let balance = get_near_balance(&master_account, CONTRACT_ACC);
+    assert_eq!(balance, DEPOSITED_FEE);
+
+    let balance = total_supply(&master_account);
+    assert_eq!(balance, DEPOSITED_AMOUNT - withdraw_amount as u128);
+}
+/*
+#[test]
+fn test_withdraw_eth() {
```

Comment:
Should this test still be commented out?

Reply:
@birchmd that method will be removed, as Alex Shevchenko announced. So we don't need that test.